From 9dc3562f9505ac7af4054fddddff229ffc8e889e Mon Sep 17 00:00:00 2001
From: Chen Jun <chenjun102@huawei.com>
Date: Mon, 1 Mar 2021 02:34:03 +0000
Subject: [PATCH] rockchip: Make cdn_dp_resume depend on CONFIG_PM_SLEEP
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 9dc3562f9505ac7af4054fddddff229ffc8e889e

MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

hulk inclusion
category: bugfix
bugzilla: 50067
CVE: NA

-------------------------------------------------------------------------

If build Image without CONFIG_PM_SLEEP, there would be a compile warning:
warning: ‘cdn_dp_resume’ defined but not used [-Wunused-function]

Because SET_SYSTEM_SLEEP_PM_OPS will do nothing without CONFIG_PM_SLEEP.

Make cdn_dp_resume depend on CONFIG_PM_SLEEP

Signed-off-by: Chen Jun <chenjun102@huawei.com>
Reviewed-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Chen Jun <chenjun102@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/gpu/drm/rockchip/cdn-dp-core.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/gpu/drm/rockchip/cdn-dp-core.c
+++ b/drivers/gpu/drm/rockchip/cdn-dp-core.c
@@ -1122,6 +1122,7 @@ static int cdn_dp_suspend(struct device
 	return ret;
 }
 
+#ifdef CONFIG_PM_SLEEP
 static __maybe_unused int cdn_dp_resume(struct device *dev)
 {
 	struct cdn_dp_device *dp = dev_get_drvdata(dev);
@@ -1134,6 +1135,7 @@ static __maybe_unused int cdn_dp_resume(
 
 	return 0;
 }
+#endif
 
 static int cdn_dp_probe(struct platform_device *pdev)
 {
